### PR TITLE
fix shellcheck failures in ./hack/jenkins/...

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
fix shellcheck failures in ./hack/jenkins/...

**Which issue(s) this PR fixes**:

Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xichengliudui** [comment]
Why did you modify and fix what? I wrote it very clearly.
https://github.com/kubernetes/kubernetes/pull/75865/files

**Review by neolit123 — COMMENTED** [review]
LGTM
/priority backlog
/sig testing

**Comment by xichengliudui** [comment]
Here.

**Comment by SataQiu** [comment]
Thanks @xichengliudui 
/lgtm

File: hack/jenkins/benchmark-dockerized.sh [hunk 1]
```diff
@@ -51,5 +56,5 @@ cd /go/src/k8s.io/kubernetes
 # Run the benchmark tests and pretty-print the results into a separate file.
 make test-integration WHAT="$*" KUBE_TEST_ARGS="-run='XXX' -bench=. -benchmem" \
   | tee \
-   >(prettybench -no-passthrough > ${ARTIFACTS}/BenchmarkResults.txt) \
-   >(go run test/integration/benchmark/jsonify/main.go ${ARTIFACTS}/BenchmarkResults_benchmark_$(date -u +%Y-%m-%dT%H:%M:%SZ).json || cat > /dev/null)
+   >(prettybench -no-passthrough > "${ARTIFACTS}/BenchmarkResults.txt") \
+   >(go run test/integration/benchmark/jsonify/main.go "${ARTIFACTS}"/BenchmarkResults_benchmark_"$(date -u +%Y-%m-%dT%H:%M:%SZ)".json || cat > /dev/null)
```
Comment (fejta) [inline hack/jenkins/benchmark-dockerized.sh]: Can you quote the whole string not just the variable part?

File: hack/jenkins/verify-dockerized.sh [hunk 2]
```diff
@@ -21,7 +21,12 @@ set -o xtrace
 
 retry() {
   for i in {1..5}; do
-    "$@" && return 0 || sleep "${i}"
+    if "$@"
```
Comment (fejta) [inline hack/jenkins/verify-dockerized.sh]: Typically we use a ; and put if and then on the same line.

**Review by fejta — COMMENTED** [review]
/approve

**Pull request merged**